tcms/core/ajax.py

```diff
@@ -398,9 +400,10 @@ def get_prod_related_objs(p_pks, target):
         'build': (Build, 'name'),
         'category': (Category, 'name'),
     }
-    results = ctypes[target][0]._default_manager.filter(product__in=p_pks)
     attr = ctypes[target][1]
-    results = [(r.pk, getattr(r, attr)) for r in results]
+    results = []
```

Comment:
Both of the `get_prod_related` functions are used only in `static/js/advance_search.js` which despite its incorrect name is used only in 1 place it seems (reporting templates). 

In particular this view is used in `getProdRelatedObj` which can be replaced with the new `jsonRPC` client. Give it a try in the browser console first to see how it goes. I think you should be able to remove (at least initially) this view and fetch the necessary information via the existing API methods.

Reply:
Playing around with this makes me feel this is not such a trivial task. First I get the `$` inside of `jsonRPC` as undefined. It works when I replace it with `jQ` but then I get other problems. Also, this method is generic and allows you to pass a `target`. Doing this directly with the javascript through the API will require some hacks on the JS side. Also, the lack of tests makes me fear we may break something with this refactoring.

How about we leave this as it is now, and I log a task for the future to investigate further and remove the methods if unnecessary?

Reply:
You are right that there is much more to that view. On the server side it duplicates `ajax.info()` and on the client side the search form and its JavScript functions duplicate `getInfo()` and all the places it is used. Also for modern search form (SEARCH menu ->) we use a bunch of utils funtions `js/utils.js` where we have functions to update a select box for builds and filters.

Mark these views with a todo comment for removal and link to this comment for reference.

About the jsonRPC issues see 27b155e47545c1df974ce0f0db3ad5020c9b0bef and the latest commits on the master branch. If you load jquery and jsonrpc.js as the last JavaScript files in the template all should be good. TP, TC (as of above commit) and TR pages (get view) already load the `jsonrpc.js` file and successfully utilize the API interface.

References to `jQ` are a problem and they should be updated to `$`. Also `tcms_actions.js` has this fallback right at the top:
```
if (!jQ) {
    var jQ = $;
}
```

This is for cases where you are using functions from `tcms_actions.js` inside templates that inherit from `tcms_base.html` instead of `base.html` (which doesn't register jQuery as the jQ variable).

Reply:
I have put a TODO comment, as discussed